bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyBaseHandler.java

```diff
@@ -61,38 +68,39 @@
 @NonNullByDefault
 public class ShellyBaseHandler extends BaseThingHandler implements ShellyDeviceListener {
     protected final Logger logger = LoggerFactory.getLogger(ShellyBaseHandler.class);
+    protected final ShellyChannelDefinitionsDTO channelDefinitions;
 
     public String thingName = "";
-    protected ShellyBindingConfiguration bindingConfig = new ShellyBindingConfiguration();
+    public String thingType = "";
+
+    protected final ShellyHttpApi api;
+    protected ShellyBindingConfiguration bindingConfig;
     protected ShellyThingConfiguration config = new ShellyThingConfiguration();
-    protected @Nullable ShellyHttpApi api;
-    private @Nullable ShellyCoapHandler coap;
-    protected @Nullable ShellyDeviceProfile profile;
-    private final @Nullable ShellyCoapServer coapServer;
+    protected ShellyDeviceProfile profile = new ShellyDeviceProfile(); // init empty profile to avoid NPE
+    private final ShellyCoapHandler coap;
+    private Boolean autoCoIoT = false;
     protected boolean lockUpdates = false;
 
-    @SuppressWarnings("unused")
-    private long lastUpdateTs = 0;
+    private final ShellyTranslationProvider messages;
+    private boolean channelsCreated = false;
+
     private long lastUptime = 0;
     private long lastAlarmTs = 0;
-    private Integer lastTimeoutErros = -1;
+    private long lastTimeoutErros = -1;
 
-    private @Nullable ScheduledFuture<?> statusJob;
-    private int skipUpdate = 0;
+    private Optional<ScheduledFuture<?>> statusJob = Optional.empty();
```

Comment:
Optional isn't really a very good choice in this case. It was better when it was a nullable field.

Reply:
change reverted
why?

Reply:
Using nullables for future fields is an unofficial openhab coding standard. We are just trying to make all the bindings consistent.

Reply:
aha